File: token-client-core/src/main/java/no/nav/security/token/support/client/core/oauth2/AbstractOAuth2GrantRequest.java
package no.nav.security.token.support.client.core.oauth2;

import no.nav.security.token.support.client.core.ClientProperties;
import no.nav.security.token.support.client.core.OAuth2GrantType;

import java.util.Objects;

abstract class AbstractOAuth2GrantRequest {

    private final OAuth2GrantType oAuth2GrantType;
    private final ClientProperties clientProperties;

    protected AbstractOAuth2GrantRequest(OAuth2GrantType oAuth2GrantType, ClientProperties clientProperties) {
        this.oAuth2GrantType = oAuth2GrantType;
        this.clientProperties = clientProperties;
    }

    OAuth2GrantType getGrantType() {
        return oAuth2GrantType;
    }

    ClientProperties getClientProperties() {
        return clientProperties;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AbstractOAuth2GrantRequest that = (AbstractOAuth2GrantRequest) o;
        return Objects.equals(oAuth2GrantType, that.oAuth2GrantType)
            && Objects.equals(clientProperties, that.clientProperties);
    }

    @Override
    public int hashCode() {
        return Objects.hash(oAuth2GrantType, clientProperties);
    }

    @Override
    public String toString() {
        return getClass().getSimpleName() + " [oAuth2GrantType=" + oAuth2GrantType + ", clientProperties=" + clientProperties + "]";
    }
}

File: token-client-core/src/main/kotlin/no/nav/security/token/support/client/core/oauth2/AbstractOAuth2GrantRequest.kt
package no.nav.security.token.support.client.core.oauth2

import com.nimbusds.oauth2.sdk.GrantType
import java.util.Objects
import no.nav.security.token.support.client.core.ClientProperties

abstract class AbstractOAuth2GrantRequest(val grantType : GrantType, val clientProperties : ClientProperties) {

    override fun equals(other : Any?) : Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as AbstractOAuth2GrantRequest
        return grantType == that.grantType && clientProperties == that.clientProperties
    }

    override fun hashCode() = Objects.hash(grantType, clientProperties)
    override fun toString() = "${javaClass.getSimpleName()} [oAuth2GrantType=$grantType, clientProperties=$clientProperties]"
}